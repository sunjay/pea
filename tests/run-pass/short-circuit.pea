/// Prints the given operand and then returns true if it is greater than 3
fn gt3(x: i64) -> bool {
    println!(x);
    x > 3
}

fn main() {
    // left runs, but right does not
    if gt3(4) || gt3(5) {
        println!(b"wow!");
    }
    if gt3(0) && gt3(16) {
        println!(b"cool!");
    }

    // both left and right run, but the body does not
    if gt3(2) || gt3(-1) {
        println!(b"aww");
    }
    if gt3(55) && gt3(-1000) {
        println!(b"bad");
    }

    // both left and right run, and the body runs too
    if gt3(-2) || gt3(6) {
        println!(b"awesome");
    }
    if gt3(99) && gt3(77) {
        println!(b"love it");
    }

    // compound conditions
    if gt3(-2) || gt3(4) && gt3(56) {
        println!(b"big fan!");
    }
}
