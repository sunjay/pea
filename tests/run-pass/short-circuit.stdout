4
wow!
0
2
-1
55
-1000
-2
6
awesome
99
77
love it
-2
4
56
big fan!
